### DISCUSSION TITLE
Unreferenced property  in customize emitter plugin

### DISCUSSION THREAD

**Discussion by yangocean-sudo** [mitsuba-renderer/mitsuba3#628] [Q&A] [answered]
Hi, [as we discussed before](https://github.com/mitsuba-renderer/mitsuba3/discussions/614), we are going to create a customise "photon emitter" that can read the input binary file and then emit the rays. 
It works fine for my local computer. It is a mac with Intel Core, and the variant I used is 
 `llvm_ad_rgb`.
Now I'm trying to test this emitter in the school remote GPU machine, and the variant I need to use is `cuda_mono`, and when I read the XML file from jupyter notebook, it shows:
```
RuntimeError                              Traceback (most recent call last)
Input In [2], in <cell line: 1>()
----> 1 scene = mi.load_file("./two_mirrors.xml")
      2 original_image = mi.render(scene)
      3 import matplotlib.pyplot as plt

RuntimeError: ​[xml.cpp:1157] Error while loading "./two_mirrors.xml" (near line 46, col 6): unreferenced property "["filename"]" in emitter plugin of type "photon_emitter"
```
I noticed that the file `envmap.cpp` is  worthy of reference, and I try to be consistent with the way it reads files.  And this is my snippet  code to read the file
```
if (props.has_property("filename"))
            std::cout << "Reading file..." <<std::endl;
// Read the file
FileResolver *fs = Thread::thread()->file_resolver();
fs::path file_path = fs->resolve(props.string("filename"));
m_filename = file_path.filename().string();
ref<FileStream> fileStream = new FileStream(file_path, FileStream::ERead);
```
Did I miss anything? Thank you for your time.

**Comment by njroussel** [comment, marked as answer]
Hi @yangocean-sudo 

No, this all seems correct. Since this code also works on your local machine, this sounds like an environment issue on the remote machine. Check the mitsuba path with `import mitsuba as mi; mi.__path__` to be sure that it is pointing to your custom build.

**Reply by yangocean-sudo** [reply]
Thank you! The problem is solved by checking the mitsuba path.
